trojsten/submit/views.py

```diff
@@ -215,6 +217,68 @@ def active_rounds_submit_page(request):
     )
 
 
+@login_required
+def all_submits_page(request, submit_type, article, *args, **kwargs):
+    print(kwargs)
+    submits_query_set = Submit.objects.filter(
+        user=request.user, submit_type=submit_type,
+        task__round__semester__competition__sites__in=[settings.SITE_ID],
+    ).select_related(
+        'task__round', 'task__round__semester', 'task__round__semester__competition',
+    ).order_by(
+        'task__round__semester__competition', '-task__round__semester__year', '-task__round__semester__number',
+        '-task__round__number', 'task__number', 'submit_type', '-time'
+    ).distinct(
+        'task__round__semester__competition', 'task__round__semester__year', 'task__round__semester__number',
+        'task__round__number', 'task__number', 'submit_type',
+    )
+    competitions = [None]
+    semesters = {}
+    all_submits = {}
+    for submit in submits_query_set:
+        competition = submit.task.round.semester.competition
+        semester = submit.task.round.semester
+        if competition != competitions[-1]:
+            competitions.append(competition)
+            all_submits[competition] = {}
+            semesters[competition] = [None]
+        if semester != semesters[competition][-1]:
+            semesters[competition].append(semester)
+            all_submits[competition][semester] = {}
+            for round in semester.round_set.all():
+                all_submits[competition][semester][round] = []
+        all_submits[competition][semester][submit.task.round].append(submit)
+
+    kwargs.update({
+        'article': article,
+        'all_submits': all_submits,
+        'competitions': competitions[1:],
+        'all_semesters': semesters,
+        'IN_QUEUE': constants.SUBMIT_STATUS_IN_QUEUE,
+        'submit_type': submit_type,
+        'DESCRIPTION': constants.SUBMIT_TYPE_DESCRIPTION,
+    })
+
+    return render(
+        request,
+        'trojsten/submit/all_submits_page.html',
+        kwargs
+    )
+
+
+@login_required
+@get_article(can_read=True)
```

Comment:
Preco tuna (a nizsie) potrebujeme article? Nejde to bez toho?

Reply:
Chcem to mať prepojené s článkom. Príde mi to lepšie ako hardcodenutý obsah, ktorý možno iné semináre budú chcieť mať iný. Ale ide to presunúť do metódy vyššie, tak som to dal tam.

Reply:
Aky obsah tam chceme mat? Nestaci mat nadpis?

Reply:
Myslel som, že tam tak napísať, že tu môžeš vidieť všetky úlohy, možno aj nejaké ďalšie užitočné informácie, ale to vie byť niekde inde ako tu. Takže myslím, že ten článok na koniec až tak netreba.